src/SRSMain.cpp

```diff
@@ -28,128 +29,91 @@ SRSMain::Reprocess(){
-              eventBuilder = new SRSEventBuilder(nEvent,_conf->GetMaxClusterSize(), _conf->GetMinClusterSize(), _conf->GetZeroSupCut(), _conf->GetRunType(), 
-						 std::string(_conf->GetClusterPositionCorrectionFlag()).find("applyCorrections")!=std::string(_conf->GetClusterPositionCorrectionFlag()).npos);
-            }
-	    if(debug_s)	std::cout << " >>>> BAD! Merging two fragment in  "<<nEvent<< " tFEC "<<iFEC<<" # word "<<previous.size()<<std::endl;
-	    //decode previous                                                                                                                                                                            
-            for(unsigned int i=0;i<previous.size();i++){
-              event[i]=previous[i];
-            }
-	    if (debug_s) std::cout <<" +++ reprocess: event # "<<nEvent<<" FEC "<<tFEC<<std::endl;
-            SRSFECDecoder dec(eventBuilder);
-            dec.decodeFEC(previous.size(),event);
-            previous.clear();
-          }	  
-        }else{
-	  if(debug_s) std::cout <<" >>>> SKIP the decoding waiting next fragment"<<std::endl;
-          for (unsigned int ir=0; ir < nw-start_eventRecord;ir++){
-            previous.push_back(buffer[start_eventRecord+ir]);
-          }
-          prenevt=nEvent;
+        nadc++;
+        int adcId = j&0xff;//(( j >>8 ) & 0xffffffff);                                                                                  
+	if (debug_s)std::cout <<"SRSMain::  +++ reprocess: "<<std::hex<<" ADC Id = 0x"<<adcId<<std::dec<<std::endl;
+	//	goaway = (nadc == 35);
+      } else if ((j&0xffffffff) == 0xfafafafa){
+        firstADCinFEC = true;
+        fillFECdata = false;
+        iw++;
+        buffer[iw]=j;
+        if (iFEC == 1) {
+	  eventBuilder = new SRSEventBuilder(nEvent,_conf->GetMaxClusterSize(), _conf->GetMinClusterSize(), _conf->GetZeroSupCut(),\
+					     _conf->GetRunType(),
+					     std::string(_conf->GetClusterPositionCorrectionFlag()).find("applyCorrections")!=std::\
+					     string(_conf->GetClusterPositionCorrectionFlag()).npos);
+
+          nEvent++;
+	  if(debug_s)std::cout <<"New Event "<<nEvent<<std::endl;
         }
-        if (tFEC == nFEC){
-          // The event is now complete and we can form the cluster
+        if (iFEC == nFEC){
+	  SRSFECDecoder dec(eventBuilder);
+	  dec.decodeFEC(iw,buffer);
+	  if (nEvent%1000==0) std::cout<<" SRSMain:: +++ reprocess: Event # "<<nEvent<<std::endl;
 	  if (debug_s) std::cout <<" +++ reprocess: compute cluster"<<std::endl;
           eventBuilder->ComputeClustersInDetectorPlane();
-          // Fill the trees
+          // Fill the trees                                                                                                             
           _root->FillRootFile(eventBuilder);
           eventBuilder=0;
-	  
-        }
-        if (iFEC == 2) {
-          newevent = true;
-          //  delete eventBuilder;
-          iFEC = 0;
+
+        }else if (iFEC > 3){
```

Comment:
I'm using a map<FecId, AdcId> but the concept is the same and once this is ok we can drop the else if statement in toto.. since the idea is to fill this looking at the data, and compare with a similar one coming from the config, which is available....  as it stands now the code work with 3 FECs... so the idea is to ask Marek to  make few runs  with 3 FECs and look at the physics content to see if it is ok.